lib/src/native/realm_core.dart

```diff
@@ -1665,14 +1692,13 @@ class _RealmCore {
     });
   }
 
-  static void _logCallback(Object userdata, int levelAsInt, Pointer<Int8> message) {
-    final logger = Realm.logger;
-    final level = LevelExt.fromInt(levelAsInt);
+  void loggerLogMessage(int level, String message) {
+    final message_level = LevelExt.fromInt(level);
+    Realm.logger.log(message_level, message);
+  }
 
-    // Don't do expensive utf8 to utf16 conversion unless needed.
-    if (logger.isLoggable(level)) {
-      logger.log(level, message.cast<Utf8>().toDartString());
-    }
+  void loggerSetLogLevel(Level logLevel) {
```

Comment:
```suggestion
  void setLogLevel(Level logLevel) {
```

Reply:
This name `loggerSetLogLevel` was suggested by @blagoev above.